regex URL converter matches incorrect value

I have the following code in a Flask App using Werkzeug

```py

# Redirect requests for the older image URLs to new URLs
@app.route(
    '/static/images/2019/<regex("(privacy|jamstack|capabilities)"):folder>/<image>'
)
def redirect_old_hero_images(folder, image):
    return redirect("/static/images/2020/%s/%s" % (folder, image)), 301
```

prior to 2.2.0 this would allow the following redirect:

```
/static/images/2019/jamstack/random.png
```

to

```
/static/images/2020/jamstack/random.png
```

Now however it incorrectly sends us to:

```
/static/images/2020/jamstack/jamstack
```

Pinning werkzeug to 2.1.2 fixes the issue.

Environment:

- Python version: 3.8
- Werkzeug version: 2.2.0 and 2.2.1

Apologies for that, wasn't aware that this wrapped some other code.

Here's a minimal reproducible pytest example which works in 2.1.2 but not in 2.2.0 nor 2.2.1:

```py
from werkzeug.routing import BaseConverter

class RegexConverter(BaseConverter):
    def __init__(self, url_map, *items):
        super(RegexConverter, self).__init__(url_map)
        self.regex = items[0]

def test_regex():
    m = r.Map(
        [
            r.Rule("/<regex('(privacy|jamstack|capabilities)'):folder>/<image>", endpoint="simple"),
        ], converters={'regex': RegexConverter}
    )
    a = m.bind("localhost", path_info="/jamstack/random.png")

    assert a.match() == ("simple", {"folder": "jamstack", "image": "random.png"})
```

Result from 2.2.1:

```
>       assert a.match() == ("chapter_image", {"folder": "jamstack", "image": "random.png"})
E       AssertionError: assert ('chapter_ima...: 'jamstack'}) == ('chapter_ima...'random.png'})
E         At index 1 diff: {'folder': 'jamstack', 'image': 'jamstack'} != {'folder': 'jamstack', 'image': 'random.png'}
E         Use -v to get more diff
```

You can see it's incorrectly providing the `folder` match (`jamstack`) to both the `folder` AND the `image` variable.

--- a/docs/routing.rst
+++ b/docs/routing.rst
@@ -105,10 +105,6 @@ converters can be overridden or extended through :attr:`Map.converters`.
 
 .. autoclass:: UUIDConverter
 
-If a custom converter can match a forward slash, ``/``, it should have
-the attribute ``part_isolating`` set to ``False``. This will ensure
-that rules using the custom converter are correctly matched.
-
 
 Maps, Rules and Adapters
 ========================
@@ -168,7 +164,13 @@ built-in converters. To make a custom converter, subclass
 
 The converter should have a ``regex`` attribute with a regular
 expression to match with. If the converter can take arguments in a URL
-rule, it should accept them in its ``__init__`` method.
+rule, it should accept them in its ``__init__`` method. The entire
+regex expression will be matched as a group and used as the value for
+conversion.
+
+If a custom converter can match a forward slash, ``/``, it should have
+the attribute ``part_isolating`` set to ``False``. This will ensure
+that rules using the custom converter are correctly matched.
 
 It can implement a ``to_python`` method to convert the matched string to
 some other object. This can also do extra validation that wasn't

